### PR TITLE
fix(combobox, dropdown, input-date-picker, popover, tooltip): disable scroll event listeners for hidden poppers

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** #3727 

## Summary

This PR updates all popper-owning components to toggle the `eventListeners` modifier based on the popper's visibility (as suggested [here](https://popper.js.org/docs/v2/tutorial/#performance)). 

This also cleans up some internals for combobox. It seemed to have multiple state props based on `active`.

### CODE REVIEW

**Comment by driskull** [comment]
nice! Thanks for this Franco.

File: src/components/combobox/combobox.tsx [hunk 1]
```diff
@@ -1065,7 +1058,7 @@ export class Combobox implements LabelableComponent, FormComponent, InteractiveC
   }
 
   render(): VNode {
-    const { guid, open, label } = this;
```
Comment (driskull) [inline src/components/combobox/combobox.tsx:1068]: Looks like `open` was being used below. Causing a JS error.

**Review by driskull — CHANGES_REQUESTED** [review]

File: src/components/combobox/combobox.tsx (hunk 1 above)
Comment (jcfranco) [reply src/components/combobox/combobox.tsx:1068]: Thanks for catching this. There were other references that I missed that didn't come up in focused tests.

**Review by driskull — APPROVED** [review]
👍 Nice!

**Pull request merged**